configuration/scripts/ratos-logging.sh

```
@@ -0,0 +1,311 @@
+
+    return "$exit_code"
+}
+
+# Function to set up error trapping
+setup_error_trap() {
+    local context="$1"
+
+    # Enable error trapping but be more selective about exit behavior
+    set -E  # Inherit ERR trap to functions and subshells
+
+    trap "handle_error \$? \$LINENO \"${context}\"" ERR
```

Shellcheck (0.10.0)

[warning] Use single quotes, otherwise this expands now rather than when signalled.

(SC2064)